lib/src/rules/public_member_api_docs.dart

```diff
@@ -101,9 +104,21 @@ class Visitor extends GeneralizingAstVisitor {
   bool isOverridingMember(Declaration node) =>
       getOverriddenMember(node.element) != null;
 
+  /// Return true if the given node is declared in a compilation unit that is in
+  /// a `lib/` folder.
+  bool isDefinedInLib(AstNode node) {
+    CompilationUnit compilationUnit = node is CompilationUnit
+        ? node
+        : node.getAncestor((node) => node is CompilationUnit);
+    Uri uri = compilationUnit?.element?.source?.uri;
+
+    // TODO(devoncarew): This will have false positives.
+    return uri != null && uri.pathSegments.contains('lib');
```

Comment:
This is assuming that all the compilation units this lint will encounter are in a pub-style package. That isn't necessarily true.

But for unit inside a pub-style package, do we want to exclude compilation units in 'lib/src'?

Reply:
> This is assuming that all the compilation units this lint will encounter are in a pub-style package. That isn't necessarily true.

Perhaps this lint could only be used for those projects though? As in, it doesn't make sense to enable in a project that doesn't use pub conventions.

> But for unit inside a pub-style package, do we want to exclude compilation units in 'lib/src'?

That's very tricky. The issue I'm seeing now is that this lint is reporting violations for directories like `bin/` and `test/` - that can never export their APIs. That's what this PR is trying to solve.

Excluding `lib/src/` is technically correct, but a much trickier issue, since some compilation units in there could be exported by files in `lib/`. Having a correct view of the public api of a package - and ensuring that all public members are documented - is a good goal, but a bit outside the scope of what I want to do here.

Reply:
> Perhaps this lint could only be used for those projects though? As in, it doesn't make sense to enable in a project that doesn't use pub conventions.

I don't see why it wouldn't make sense to enable this rule in a non-pub package, but if we want to restrict it to only work in pub packages we should document that it is only intended to be used in that context (or the relevant restriction, such as "only lints files inside a 'lib' directory").